ui/src/services/alertUtils.js

```diff
@@ -0,0 +1,208 @@
+//@flow
+import {
+  STATUS_CRITICAL,
+  STATUS_WARNING,
+  STATUS_HEALTH,
+  STATUS_NONE,
+} from '../constants';
+import { getAlertsLoki } from './loki/api';
+import type { PrometheusAlert, AlertList, Health } from './alertmanager/api';
+import { compareHealth } from './utils';
+
+// Return boolean if the two alerts have the same labels but different severity
+const isSameAlertWithDiffSeverity = (
+  alert1: {
+    [labelName: string]: string,
+    parents?: string[],
+    selectors?: string[],
+  },
+  alert2: {
+    [labelName: string]: string,
+    parents?: string[],
+    selectors?: string[],
+  },
+): boolean => {
+  // filter out the `severity` property
+  function replacer(key, value) {
+    if (key === 'severity') {
+      return undefined;
+    }
+    return value;
+  }
+
+  return JSON.stringify(alert1, replacer) === JSON.stringify(alert2, replacer);
+};
+
+/* 
+This function removes the warning alert from the list when critical one is triggered
+It should be called at saga level before storing the alerts to `redux-store`
+or where we resolve the promise with `react-query` 
+*/
+export const removeWarningAlerts = (alerts: AlertList): AlertList => {
+  const len = alerts.length;
+  const removeIndex = [];
+  for (let i = 0; i < len - 1; i++) {
+    for (let j = i + 1; j < len; j++) {
+      if (isSameAlertWithDiffSeverity(alerts[i].labels, alerts[j].labels)) {
+        if (alerts[i].labels.severity === STATUS_WARNING) {
+          removeIndex.push(i);
+        } else if (alerts[j].labels.severity === STATUS_WARNING) {
+          removeIndex.push(j);
+        }
+      }
+    }
+  }
+
+  let removedWarningAlerts = [...alerts];
+  removeIndex.forEach((index) => removedWarningAlerts.splice(index, 1));
+
+  return removedWarningAlerts;
+};
+
+// Sort the alerts base on the `severity`
+export const sortAlerts = (alerts: AlertList): AlertList => {
+  return alerts.sort(function (a, b) {
+    return compareHealth(a.labels.severity, b.labels.severity);
+  });
+};
+
+export const isObjectHasProperty = (obj: {}, property: string): boolean => {
+  return Object.prototype.hasOwnProperty.call(obj, property);
+};
```

Comment:
I'm not sure this method is necessary. as label values can only be of type string, wouldn't it be easier to use `!!obj.property` ?